### PR TITLE
Add testing for Go Tip and Go 1.16 and get rid of Travis CI and Go 1.14.

### LINKED ISSUE #548: Remove Travis CI once Github Action supports Go master branch.
Remove Travis CI once https://github.com/actions/setup-go/pull/34 is implemented so that we can move everything into Github Action.

### PR DESCRIPTION
**Pull request by wyk9787**
Fixes #548.

I spent over an hour on it but was not able to make Go Tip for Windows work. Since it's not high priority for Windows support, this PR does not add Go Tip testing on Windows.

Since macOS and Xcode requirements are convoluted (see https://developer.apple.com/support/xcode/), I am currently testing the following:
* The latest minor version on the latest major version (i.e. 12.x)
* The latest minor version on the previous major version (i.e. 11.7)

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -18,14 +18,29 @@ jobs:
         working-directory: ${{ env.WORKING_DIR }}
     strategy:
       matrix:
-        go: ['1.14', '1.15']
```
Comment (nolanmar511) [inline .github/workflows/ci.yaml:21]: Mention eliminating 1.14 in description?

**Review by nolanmar511 — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (wyk9787) [reply .github/workflows/ci.yaml:21]: Updated the PR title.

**Review by nolanmar511 — APPROVED** [review]

**Comment by nolanmar511** [comment]
@aalexand -- Do you have permission to modify the expected set of tests? (I think it's under Settings > Branches > Branch protection rules)

**Comment by wyk9787** [comment]
> @aalexand -- Do you have permission to modify the expected set of tests? (I think it's under Settings > Branches > Branch protection rules)

Thanks! I was about to ask that :)

@aalexand, could you disable tests that are removed from this PR (tests that are showing "Expected - Waiting for status to be reported") and add tests that are newly added in this PR as requirements (not sure if they already appear in the Settings; if not, they should after the PR got submitted).

**Comment by aalexand** [comment]
> @aalexand, could you disable tests that are removed from this PR (tests that are showing "Expected - Waiting for status to be reported") and add tests that are newly added in this PR as requirements (not sure if they already appear in the Settings; if not, they should after the PR got submitted)? Thanks!

@nolanmar511 I made you an admin of the repo so that you can make these changes too and so that I'm not a SPOF. So you should be able to address this yourself, but please LMK if you can't or if you still want me to do this.

**Comment by nolanmar511** [comment]
Awesome! Looks like I do have permissions to change things. Updated the required checks.

**Pull request merged**